### PR TITLE
Report Card work done

### PR DESCRIPTION
**Pull request by LeahBineth**

### CODE REVIEW

File: 1 tables.sql [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+create table dbo.ReportCard(
+	ReportCardId int not null identity primary key,
+	FirstName varchar(30) not null 
+		constraint ck_ReportCard_StudentFirstName_cannot_be_blank check(FirstName > ''),
+	LastName varchar(30) not null 
+		constraint ck_ReportCard_StudentLastName_cannot_be_blank check(LastName > ''),
+	DateBorn date not null 
+		constraint ck_ReportCard_DateBorn_cannot_be_in_the_future check(DateBorn < getdate()),
+	PhoneNumber varchar(15) not null 
+		constraint ck_ReportCard_PhoneNumber_cannot_be_blank check(PhoneNumber > ''),
```
Comment (MKlugman) [inline 1 tables.sql]: PhoneNumber should only be 10 digits and check that's it's only numbers inserted....

File: 1 tables.sql [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+		constraint ck_ReportCard_ScienceScore_must_be_between_45_and_100 check(ScienceScore between 45 and 100),
+	HistoryScore int not null 
+		constraint ck_ReportCard_HistoryScore_must_be_between_45_and_100 check(HistoryScore between 45 and 100),
+	EnglishScore int not null 
+		constraint ck_ReportCard_EnglishScore_must_be_between_45_and_100 check(EnglishScore between 45 and 100),
+	GymScore varchar(2) not null 
+		constraint ck_ReportCard_GymScore_must_be_either_s_s_plus_s_minus check(GymScore in('S', 'S+', 'S-')),
+	ConductScore varchar(2) not null 
+		constraint ck_ReportCard_ConductScore_must_be_either_s_s_plus_s_minus check(ConductScore in('S', 'S+', 'S-')),
+	--AvgScore as ((MathScore + ScienceScore + HistoryScore + EnglishScore) / 4),
```
Comment (MKlugman) [inline 1 tables.sql]: AvgScore should be added as column to table.

File: 3 reports.sql [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+--3) Our school gives an award for the 4 students with the highest averages that also receive an S+ in Conduct. 
+--	 Please list the 4 highest averages in our school. (Can be more than 1 per grade)
+--	 Please include Name, Grade, Class, and all test scores in this report. 
+select top 4 
+	r.FirstName, 
+	r.LastName, 
+	r.Grade, 
+	r.ClassroomNumber, 
+	GeneralAvg = ((MathScore + ScienceScore + HistoryScore + EnglishScore) / 4),
```
Comment (MKlugman) [inline 3 reports.sql]: if you add GeneralAvg as column, then you don't have to do it manually here or by rest of reports....

File: 3 reports.sql [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+--    Please list all students in the 9th grade with this code for their Math test score.
+--    And please include a list of all students in the 10th grade with this code for their English test score. 
+select 
+	r.Grade, 
+	StudentCode =  CONCAT(right(r.PhoneNumber, 4), LEFT(r.lastname,  1), LEFT(r.firstname, 1)),  
+	TestScoreTitle = case r.grade when 9 then 'Math Score' when 10 then 'English Score' end,
+	TestScore = case r.Grade when 9 then r.MathScore when 10 then r.EnglishScore end 
+from ReportCard r
+where r.Grade in(9, 10)
+
```
Comment (MKlugman) [inline 3 reports.sql:49]: Either sort this by grade or do as 2 separate reports.

**Review by MKlugman — CHANGES_REQUESTED** [review]

**Comment by MKlugman** [comment]
I reviewed the work. Thank you! Please see few comments.

File: 3 reports.sql (hunk 4 above)
Comment (LeahBineth) [reply 3 reports.sql:49]: LB: I have a `order by r.Grade`... Isn't this what you wanted?

**Comment by LeahBineth** [comment]
LB: See corrections and one comment... Good luck!

File: 3 reports.sql (hunk 4 above)
Comment (MKlugman) [reply 3 reports.sql:49]: Yes Please include `order by r.Grade`
Thank you!

**Pull request closed without merging**